modules/extension/ysld/src/main/java/org/geotools/ysld/encode/YsldEncodeHandler.java

```diff
@@ -225,6 +225,17 @@ Object toObjOrNull(Expression expr, boolean isname) {
             return null;
 
         List<Expression> subExpressions = Util.splitConcatenates(expr);
+
+        if (subExpressions.size() == 1) {
+            Expression subExpr = subExpressions.get(0);
+            if (subExpr instanceof Literal) {
+                Object obj = ((Literal) subExpr).getValue();
+                if (!(obj instanceof String)) {
+                    return obj;
+                }
+            }
+        }
+
```

Comment:
This seems quite brittle.
Tracing through the existing YSLD logic, it looks like the only time this gets called with a Color an an argument is when it is called by toColorOrNull (line 170 of this class). Rather than risk breaking the behaviour of toObjOrNull, it would be better to modify that method to return the Color object instead.

A simple check prior to calling this method should be sufficient (replace line 170 with the following):

```
Object obj;
if (expr instanceof Literal) {
  obj = ((Literal)expr).getValue();
  if (obj instanceof Color) {
    return obj;
  }
}
obj = toObjOrNull(expr, false);
```

(Additionally, this change violates the api defined in the javadocs for this method - it should return either a String or a Number; if you modify toColorOrNull instead you wont have that issue, as that method is expected to return a Color)